### PR TITLE
Remove superfluous `room_memberships` join from background update

### PR DESCRIPTION
**Pull request by MadLittleMods**
Remove superfluous `room_memberships` join from background update

Spawning from https://github.com/matrix-org/synapse/pull/15731

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [ ] ~~Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)~~
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: synapse/storage/databases/main/roommember.py [hunk 1]
```diff
@@ -1461,7 +1461,6 @@ def add_membership_profile_txn(txn: LoggingTransaction) -> int:
                 SELECT stream_ordering, event_id, events.room_id, event_json.json
                 FROM events
                 INNER JOIN event_json USING (event_id)
-                INNER JOIN room_memberships USING (event_id)
```
Comment (MadLittleMods) [inline synapse/storage/databases/main/roommember.py:1464]: I don't see any columns used from this table in the query

Comment (MadLittleMods) [reply synapse/storage/databases/main/roommember.py:1464]: This does make sure we only look at membership events but we also use `type = 'm.room.member'` below which does the same thing.

**Review by DMRobertson — APPROVED** [review]
Seems sane. I wonder if this background job is redundant now, though.

**Comment by MadLittleMods** [comment]
Thanks for the review @DMRobertson 🐭

It probably doesn't matter anymore but just showed up in the results :shrug:. I am assuming we could remove a lot of background updates after the constraints are added and the schema compatibility version gets bumped.

**Pull request merged**